packages/flutter_tools/lib/src/ios/fallback_discovery.dart

```diff
@@ -184,7 +197,7 @@ class FallbackDiscovery {
       // tool waits for a connection to be reasonable. If the vmservice cannot
       // be connected to in this way, the mDNS discovery must be reached
       // sooner rather than later.
-      await Future<void>.delayed(const Duration(seconds: kDelaySeconds));
+      await _delay(const Duration(seconds: kDelaySeconds));
```

Comment:
I don't love this pattern, it seems very one-off, but I don't really have a better suggestions.
@zanderso what do you think of this pattern?

Reply:
There's precedent for using `FakeAsync` to drive time from tests. We should be giving that a try first before resorting to something like this. If `FakeAsync` is still too heavyweight for the test for some reason, another alternative would be to create a custom `Zone` with `createTimer` callback in its `ZoneSpecification` to wrap the operation, but that could also be error prone if the `Zone` doesn't capture the right set of stuff.

Reply:
This seems pretty straightforward compared to creating a Zone (or dealing with zones at all).

The issue with fakeAsync is that the test still needs to manually elapse time. As written, the ProtocolDiscovery doesn't expose a hook so that a test could know when it is blocking and needs the FakeAsync to elapse again